### PR TITLE
feat: Add more flexible errors

### PR DESCRIPTION
**Pull request by rsmayda**
Description of changes:
- With many potential interface versions we need a consistent way to identify our errors
- This follows a paradigm set by moment.js
https://stackoverflow.com/questions/41587865/using-instanceof-on-objects-created-with-constructors-from-deep-npm-dependenci/41592087#41592087

This fixes the current problem of if the sub-packages depend on differing versions of the interface

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fhirConfig.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export interface Auth {
 
 export interface Server {
     url: string;
+    hasCORS?: boolean;
```
Comment (carvantes) [inline src/fhirConfig.ts]: What does this mean? CORS is not a boolean concept. It requires configuration parameters, most notably the allowed origins (for `Access-Control-Allow-Origin` header)

Comment (rsmayda) [reply src/fhirConfig.ts]: Yep your right -- my mistake this is in conjunction with this PR: https://github.com/awslabs/fhir-works-on-aws-routing/pull/14/files

File: src/errors/InvalidResourceError.ts [hunk 2]
```diff
@@ -5,10 +5,16 @@
...
+
     constructor(message = 'Invalid Resource') {
         // Node Error class requires passing a string message to the parent class
         super(message);
         Object.setPrototypeOf(this, InvalidResourceError.prototype);
+        this.isInvalidResourceError = true;
         this.name = this.constructor.name;
     }
 }
+export function isInvalidResourceError(err: InvalidResourceError) {
```
Comment (carvantes) [inline src/errors/InvalidResourceError.ts]: These kind of functions should be type guards

https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

**Review by carvantes — CHANGES_REQUESTED** [review]

**Comment by carvantes** [comment]
Can we create a new PR for this instead of reusing the CORS one? It's weird to have CORS commits that now have nothing to do with the Error changes. PRs are forever the accompanying documentation of commits that we merge to mainline.

**Comment by rsmayda** [comment]
Error PR -> https://github.com/awslabs/fhir-works-on-aws-interface/pull/20/files
CORS PR -> https://github.com/awslabs/fhir-works-on-aws-routing/pull/18/files

**Pull request closed without merging**